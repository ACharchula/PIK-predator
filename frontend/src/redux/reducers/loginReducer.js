import {
     LOGIN_SUCCESS, LOGIN_FAILURE, LOGOUT_SUCCESS
} from '../actions';

export const initialState = {
    login_status : "logged_out"
};

export default function loginReducer(state = initialState, action) {

    switch (action.type) {
        case LOGIN_SUCCESS:
            return {
                ...state,
                login_status: "logged_in"
            };
        case LOGIN_FAILURE:
            return {
                ...state,
                login_status: "logged_out"
            };
        case LOGOUT_SUCCESS:
            return {
                ...state,
                login_status: "logged_out"
            };
        default:
            return {
                ...state
            };
    }
}
