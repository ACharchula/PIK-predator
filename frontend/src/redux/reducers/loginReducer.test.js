import loginReducer from './loginReducer';
import {initialState} from './loginReducer';
import {
    LOGIN_SUCCESS, LOGIN_FAILURE, LOGOUT_SUCCESS
} from '../actions';


describe('login reducer', () => {
    it('should return the initial state', () => {
        expect(loginReducer(undefined, {})).toMatchSnapshot()
    });

    it('should handle LOGIN_SUCCESS', () => {
        expect(
            loginReducer(initialState,
                {
                    type: 'LOGIN_SUCCESS'
                })
        ).toMatchSnapshot()
    });

    it('should handle LOGIN_FAILURE', () => {
        expect(
            loginReducer(initialState,
                {
                    type: 'LOGIN_FAILURE'
                })
        ).toMatchSnapshot()
    });

    it('should handle LOGOUT_SUCCESS', () => {
        expect(
            loginReducer(initialState,
                {
                    type: 'LOGOUT_SUCCESS'
                })
        ).toMatchSnapshot()
    })
});