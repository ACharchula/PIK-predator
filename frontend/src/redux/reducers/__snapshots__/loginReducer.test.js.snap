// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login reducer should handle LOGIN_FAILURE 1`] = `
Object {
  "login_status": "logged_out",
}
`;

exports[`login reducer should handle LOGIN_SUCCESS 1`] = `
Object {
  "login_status": "logged_in",
}
`;

exports[`login reducer should handle LOGOUT_SUCCESS 1`] = `
Object {
  "login_status": "logged_out",
}
`;

exports[`login reducer should return the initial state 1`] = `
Object {
  "login_status": "logged_out",
}
`;
